service/matching/tasklist/matcher.go

```diff
@@ -410,8 +410,19 @@ func (tm *taskMatcherImpl) Poll(ctx context.Context, isolationGroup string) (*In
 	ctxWithCancelPropagation, stopFn := ctxutils.WithPropagatedContextCancel(ctx, tm.cancelCtx)
 	defer stopFn()
 
+	var task *InternalTask
+	var err error
+	defer func() {
+		if task != nil {
+			task.AutoConfigHint = &types.AutoConfigHint{
+				EnableAutoConfig:   tm.config.EnableClientAutoConfig(),
+				PollerWaitTimeInMs: time.Since(startT).Milliseconds(),
+			}
+		}
+	}()
+
```

Comment:
This won't work. If you need return value mutation, it needs to be initialized in the signature not just in the function scope. 
This example demonstrate the issue
https://go.dev/play/p/v11Okynpwhc

Reply:
I think this one is different since we are changing in the struct object which is reference type not a value type https://go.dev/play/p/mBOHGgaEBSW

Reply:
yea, should work fine because it's mutating.

as to doing this in a defer in general: I suppose it's no different from pushing everything into a sub-func and mutating the returned value 🤔  it does feel a bit sus / a bit error-prone at a glance though, partly due to the general defer complication @shijiesheng mentioned.  though that seems unlikely to change here, this'll likely be a pointer forever.

dunno.  not sure if we have any preference on this vs a more standard
```go
func (tm *taskMatcherImpl) Poll(ctx context.Context, isolationGroup string) {
  // ...
  task, err := pollImpl(...)
  task.AutoConfigHint = ...
  return task, nil
}
func (tm *taskMatcherImpl) pollImpl(ctx context.Context, isolationGroup string) {
  // original func
}
```
which we have all over the place.

Reply:
make sense. I didn't realize it's a pointer.